plugins/outputs/sensu/sensu.go

```diff
@@ -0,0 +1,500 @@
+		},
+		Timeout: s.Timeout.Duration,
+	}
+
+	return client, nil
+}
+
+func (s *Sensu) Connect() error {
+	if s.Timeout.Duration == 0 {
+		s.Timeout.Duration = defaultClientTimeout
+	}
+
+	if len(s.ContentEncoding) != 0 {
+		validEncoding := []string{"identity", "gzip"}
+		if !choice.Contains(s.ContentEncoding, validEncoding) {
+			return fmt.Errorf("Unsupported content_encoding [%q] specified", s.ContentEncoding)
+		}
+	} else {
+		s.ContentEncoding = "identity"
+	}
+
+	if s.BackendApiUrl != nil && s.ApiKey == nil {
+		return fmt.Errorf("backend_api_url [%q] specified, but no API Key provided", *s.BackendApiUrl)
+	}
+
+	err := s.setEndpointUrl()
+	if err != nil {
+		return err
+	}
+
+	err = s.setEntity()
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), s.Timeout.Duration)
+	defer cancel()
+
+	client, err := s.createClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	s.client = client
+
+	return nil
+}
+
+func (s *Sensu) Close() error {
+	s.client.CloseIdleConnections()
+	return nil
+}
+
+func (s *Sensu) Write(metrics []telegraf.Metric) error {
+	var points []*OutputMetric
+	for _, metric := range metrics {
+		// Add tags from config to each metric point
+		var tagList []*OutputTag
+		for name, value := range s.Tags {
+			tag := &OutputTag{
+				Name:  name,
+				Value: value,
+			}
+			tagList = append(tagList, tag)
+		}
+		for _, tagSet := range metric.TagList() {
+			tag := &OutputTag{
+				Name:  tagSet.Key,
+				Value: tagSet.Value,
+			}
+			tagList = append(tagList, tag)
+		}
+
+		// Get all valid numeric values, convert to float64
+		for _, fieldSet := range metric.FieldList() {
+			key := fieldSet.Key
+			value := getFloat(fieldSet.Value)
+			// JSON does not support these special values
+			if math.IsInf(value, 1) {
+				log.Printf("D! [outputs.sensu-go] metric %s returned positive infity, setting value to %f", key, math.MaxFloat64)
+				value = math.MaxFloat64
+			}
+			if math.IsInf(value, -1) {
+				log.Printf("D! [outputs.sensu-go] metric %s returned negative infity, setting value to %f", key, -math.MaxFloat64)
+				value = -math.MaxFloat64
+			}
+			if math.IsNaN(value) {
+				log.Printf("D! [outputs.sensu-go] metric %s returned as non a number, skipping", key)
+				continue
+			}
+
+			point := &OutputMetric{
+				Name:      metric.Name() + "." + key,
+				Tags:      tagList,
+				Timestamp: metric.Time().Unix(),
+				Value:     value,
+			}
+			points = append(points, point)
+		}
+	}
+
+	reqBody, err := s.encodeToJson(points)
+	if err != nil {
+		return err
+	}
+
+	return s.write(reqBody)
+}
+
+func (s *Sensu) write(reqBody []byte) error {
+	var reqBodyBuffer io.Reader = bytes.NewBuffer(reqBody)
+	method := http.MethodPost
+
+	if s.ContentEncoding == "gzip" {
+		rc, err := internal.CompressWithGzip(reqBodyBuffer)
+		if err != nil {
+			return err
+		}
+		defer rc.Close()
+		reqBodyBuffer = rc
+	}
+
+	req, err := http.NewRequest(method, s.EndpointUrl, reqBodyBuffer)
```

Comment:
Hmm how about using a context with timeout here using `http.NewRequestWithContext()` to prevent some blocking requests to stall the output?

Reply:
I'm not sure here and the documentation is vague in my view. So probably setting the timeout in `http.Client` is sufficient. What do you think?

Reply:
I think so.  I will remove the context bits.